src/GitVersionCore/GitPreparer.cs

```diff
@@ -83,8 +83,8 @@ public string GetProjectRootDirectory()
             log.Info($"IsDynamicGitRepository: {IsDynamicGitRepository()}");
             if (IsDynamicGitRepository())
             {
-                log.Info($"Returning Project Root as {WorkingDirectory}");
-                return WorkingDirectory;
+                log.Info($"Returning Project Root as {dynamicGitRepositoryPath}");
+                return dynamicGitRepositoryPath;
```

Comment:
@arturcic was this just overlooked? Should this method use the same `dynamicGitRepositoryPath` var that `GetDotGitDirectory()` now uses? (I'm not sure what dynamic repository means)

Reply:
AS far as I understand, `dynamicGitRepositoryPath` is the path where we can dynamically download a remote git repo and then perform the gitversion operations. @asbjornu can you confirm that? As for the use `WorkingDirectory` or `dynamicGitRepositoryPath`, the changes I did in the PRs with DI did not touch this code, it was like this before

Reply:
To be honest, this is the part of GitVersion I have the least understanding of. I think what you write is correct, @arturcic, but I believe we need to get a confirmation from @JakeGinnivan to be sure.